### ISSUE TITLE
Tmux window previewer shows blank content for all windows

### ISSUE THREAD

**Issue by akinsho** [camgraff/telescope-tmux.nvim#2]
Hi, 

Thanks for making this, it's pretty handy for tmux users 👍🏿 
One issue I've not been able to figure out is that whilst using the `window` previewer it doesn't actually show the content of my windows. So I can't tell which one I want to actually jump to

![image](https://user-images.githubusercontent.com/22454918/125750698-d669954b-9f89-4185-8ae3-2c7f24d49db4.png)

**Comment by akinsho** [comment]
Lemme know if you need more info, I'm not sure what might be useful

**Comment by camgraff** [comment]
I think it's a bug with duplicate window names in the same session. Will try to fix it in the next couple of days.

**Comment by akinsho** [comment]
~@camgraff it's probably because I have an autocommand in nvim to auto rename the window on focus gained and focus lost, so when not in focus they all have the same name 🤔😅~

EDIT: That's actually not true I just set the window name to the terminal title in my tmux config and set the terminal title to the current file in neovim

**Comment by camgraff** [comment]
I just pushed some changes. Let me know if you're still having issues

**Comment by akinsho** [comment]
@camgraff it still seems to behave the same

![image](https://user-images.githubusercontent.com/22454918/126026266-5750ccce-e455-4ae6-83dd-1de20f0aa35e.png)

![image](https://user-images.githubusercontent.com/22454918/126026324-7816f6c7-e704-4974-b957-6342c9ccb677.png)

maybe the window number is a more unique identifier, tbh I thought the status format would alter the name, just discovering that it doesn't

**Comment by camgraff** [comment]
You ran `:PlugUpdate` or `:PackerUpdate` and `:PackerCompile` if you're using packer?

If yes, I want to get clarification. The issue is the when you cycle through the entries in the "Results" window, the preview window continues to show the `.dotfiles` directory which is not the actual contents of the tmux window you are hovered over?

Also, what version of tmux are you using?

**Comment by camgraff** [comment]
> maybe the window number is a more unique identifier, tbh I thought the status format would alter the name, just discovering that it doesn't

Under the hood, it uses the session name + window id to determine which window to preview which should be unambiguous. I'll also add an option for passing a custom format string which will let you incorporate the status format in the picker entry display

**Comment by akinsho** [comment]
@camgraff I've definitely updated, and I'm on tmux version `3.0a` on Ubuntu 20.04. The thing in the screenshot that seems odd from the preview is that the tmux statusline in the example shows telescope tmux previewer as the session name, and it doesn't highlight correctly whereas the session previewer works correctly (note I had to set the `TERM` variable in telescope to get that working for sessions)

**Comment by akinsho** [comment]
Another weird thing is that it says there are 7 results which is correct but the number of results in the results' pane is not correct it shows only 6 results in the screenshot (and locally)

**Comment by camgraff** [comment]
> shows telescope tmux previewer as the session name

This is because the extension creates a "dummy session" named `telescope-tmux-previewer` and links the tmux window to the dummy session in order to display the preview.

> doesn't highlight correctly 

Can you post side by side screenshots of your tmux window normally and inside the telescope previewer to illustrate this?

> Another weird thing is that it says there are 7 results which is correct but the number of results in the results' pane is not correct it shows only 6 results in the screenshot (and locally)

The extra result is the session/window you are currently in when invoking the extension. I'll fix this.

**Comment by akinsho** [comment]
The highlighting issue is resolved for the session previewer just to be clear, I found a solution i.e. adding the terminal env variable in the telescope setup.
i.e.
```lua
  telescope.setup {
    defaults = {
      set_env = { ['TERM'] = vim.env.TERM },
    }
   }
```

### With the solution
![image](https://user-images.githubusercontent.com/22454918/126044762-ebcc55c4-4ed0-4712-880e-d3a3148a468a.png)

### Without the solution
![image](https://user-images.githubusercontent.com/22454918/126044792-6c8a652b-6b65-4313-827d-c765ee8d8b01.png)

I only mentioned it since the statusline in the previewer for the windows appears to also show a brown statusline rather than dark blue, which is expected.

### In windows previewer
![image](https://user-images.githubusercontent.com/22454918/126044840-a3169509-72f5-4178-a49c-42747fa8c7d4.png)

### In regular tmux
![image](https://user-images.githubusercontent.com/22454918/126044864-b9075c6b-0495-4795-b4eb-2234fe18885b.png)

Just to be clear this seems like a different issue from the windows not being previewed, I only mentioned just in case it provides any clues

**Comment by camgraff** [comment]
Ok, I'm still trying to understand your issue with the window previewer. So you do not have the same behavior that is shown in the gif below?
![](https://i.imgur.com/RssY4Bp.gif)

**Comment by akinsho** [comment]
Nope, that looks great. On my end it just shows an empty tmux window in the preview pane. I'll try to send you a gif when next I'm at my computer

**Comment by akinsho** [comment]
@camgraff here's a GIF of what it looks like on my machine

![tmux-telescope-bug](https://user-images.githubusercontent.com/22454918/126079626-0de6dce7-4951-47cc-9b85-179bc14fcd47.gif)

Unlike your GIF it doesn't actually show me the contents of my various tmux windows.

**Comment by camgraff** [comment]
Okay, hopefully it's fixed now. I noticed in your gif/screenshots that your previewer has 2 windows instead of just 1. I'm guessing you're setting `base-index` in your `.tmux.conf` to 1? This was resulting in the previewer showing the window at index 1 instead of 0. I changed this to take into account the `base-index` option. Changes are up, let me know if the issue persists.

**Comment by akinsho** [comment]
@camgraff that did it, it now shows the proper previews 🎉 

![image](https://user-images.githubusercontent.com/22454918/126114761-6443bd44-d6fb-4544-aab5-f963b5e65579.png)

Thanks for sticking with this 👍🏿 , one last question 😅 , passing the env variable fixed the highlighting for my sessions' previewer, but the same fix doesn't work for my windows' previewer, not sure how it works under the hood, but maybe it isn't taking the TERM env variable I'm passing into consideration.

**Comment by camgraff** [comment]
I'll look into the highlighting issue, but might take a bit. Busy lately 🙂

**Comment by akinsho** [comment]
@camgraff I've dug into this a little and I think the reason might be that
In the session previewer you use the telescope termopen previewer 
https://github.com/camgraff/telescope-tmux.nvim/blob/be1c06c396d302c7bb9046c4e90b8ed51bc3de86/lua/telescope/_extensions/tmux/sessions.lua#L69

Whereas in the window previewer you open a terminal job yourself. I think that the env of telescope is not passed to this job which is why the dummy session doesn't inherit my terminal variable.

https://github.com/camgraff/telescope-tmux.nvim/blob/be1c06c396d302c7bb9046c4e90b8ed51bc3de86/lua/telescope/_extensions/tmux/windows.lua#L60
